src/app/components/students/students.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit, Input } from "@angular/core";
+
+import { Student, Button } from "../../common/entities";
+
+import { DbService } from "../../common/services/db.service";
+
+@Component({
+  selector: "app-students",
+  templateUrl: "./students.component.html",
+  styleUrls: ["./students.component.scss"]
+})
+export class StudentsComponent implements OnInit {
+
+  private button: Button = {
```

Comment:
Good approach, but put the `class` in e.g. `enum` (`enum ButtonType`) and put that `enum` in `button.model.ts` in shared.
Actually, don't need `text` provided here. You can check in `button.component` and add corresponding text in component.